src/qililab/analog/annealing_program.py

```diff
@@ -0,0 +1,85 @@
+# Copyright 2023 Qilimanjaro Quantum Tech
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from dataclasses import dataclass
+from typing import Any, Callable
+
+import numpy as np
+
+from qililab.platform.components import Bus
+from qililab.waveforms import Arbitrary as ArbitraryWave
+
+
+@dataclass
+class AnnealingProgram:
+    """Class for an Annealing Program. The program should have the format
+
+    [
+        {"qubit_0": {"sigma_x" : 0, "sigma_y" : 1, "sigma_z" : 2},
+         "coupler_1_0 : {...},
+        },      # time=0ns
+        {...},  # time=1ns
+    .
+    .
+    .
+    ]
+    """
+
+    platform: Any
```

Comment:
The annealing program might be accessed from outside through the class object. Platform won't.

Reply:
Then it's better to define both `_platform` and `_anneal_program` as private, and  provide a public attribute `anneal_program` to read-only the annealing program (but not set it, since this is done in `__init__` only)